nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListDatabaseTables.java

```diff
@@ -0,0 +1,306 @@
+                            final String countQuery = "SELECT COUNT(1) FROM " + fqn;
+
+                            logger.debug("Executing query: {}", new Object[]{countQuery});
+                            ResultSet countResult = st.executeQuery(countQuery);
+                            if (countResult.next()) {
+                                flowFile = session.putAttribute(flowFile, DB_TABLE_COUNT, Long.toString(countResult.getLong(1)));
+                            }
+                        } catch (SQLException se) {
+                            logger.error("Couldn't get row count for {}", new Object[]{fqn});
+                            session.remove(flowFile);
+                            continue;
+                        }
+                    }
+                    if (tableCatalog != null) {
+                        flowFile = session.putAttribute(flowFile, DB_TABLE_CATALOG, tableCatalog);
+                    }
+                    if (tableSchema != null) {
+                        flowFile = session.putAttribute(flowFile, DB_TABLE_SCHEMA, tableSchema);
+                    }
+                    flowFile = session.putAttribute(flowFile, DB_TABLE_NAME, tableName);
+                    flowFile = session.putAttribute(flowFile, DB_TABLE_FULLNAME, fqn);
+                    flowFile = session.putAttribute(flowFile, DB_TABLE_TYPE, tableType);
+                    if (tableRemarks != null) {
+                        flowFile = session.putAttribute(flowFile, DB_TABLE_REMARKS, tableRemarks);
+                    }
+
+                    String transitUri;
+                    try {
+                        transitUri = dbMetaData.getURL();
+                    } catch (SQLException sqle) {
+                        transitUri = "<unknown>";
+                    }
+                    session.getProvenanceReporter().receive(flowFile, transitUri);
+                    session.transfer(flowFile, REL_SUCCESS);
+                    stateMapProperties.put(fqn, Long.toString(System.currentTimeMillis()));
+                }
+            }
+            stateManager.replace(stateMap, stateMapProperties, getScope());
+
+        } catch (final SQLException | IOException e) {
+            throw new ProcessException(e);
+        }
+    }
+
+    @Override
+    public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
+        super.onPropertyModified(descriptor, oldValue, newValue);
+        // If any of the properties that define the retrieved list have changed, then reset the state
+        if (DBCP_SERVICE.equals(descriptor)
+                || CATALOG.equals(descriptor)
+                || SCHEMA_PATTERN.equals(descriptor)
+                || TABLE_NAME_PATTERN.equals(descriptor)
+                || TABLE_TYPES.equals(descriptor)) {
+            resetState = true;
+        }
+    }
+
+    private Scope getScope() {
+        return Scope.LOCAL;
```

Comment:
Out of curiosity, wouldn't be better to have something like:
````java
private static final Scope SCOPE = Scope.LOCAL;
````
at the beginning of the class instead of this method?